challenges.yml

```diff
@@ -2088,3 +2088,217 @@ extending_prop_transition:
+          that an input element can load a piece of data.
+        c: |
+          `v-model` ensures that our input element stays in sync with the data element we
+          pass to it.
+        d: |
+          `v-model` triggers keyboard events at all times so that we can update our
+          data according to an input element's value.
+      correct: c
+
+extending_checkout_form2:
+  question: |
+    `v-model` has some "modifiers" we can add to it that change the way
+    we process data as it is entered into an input element.
+
+    Which of these options is *not* a `v-model` feature that we've seen in our
+    Vue tutorials?
+  explanation: |
+    So far we've covered `v-model.number` and `v-model.trim`. There is another
+    modifier, `v-model.lazy`, which only updates our data once the input looses
+    focus (as opposed to listening to every key up).
+
+    > Vue 3 now supports custom modifiers, which are modifiers that you can
+    > implement as functions, pretty much like *twig* and its custom filters!
+    > `v-model.capitalize` is one such example given in the docs, but it's
+    > not native to Vue!
+  features:
+    multiple_choice:
+      choices:
+        a: |
+          When `v-model.trim` is used, Vue removes all spaces before and after
+          the value entered in an input element.
+        b: |
+          When `v-model.capitalize` is used, Vue capitalizes every word entered
+          in an input element.
+        c: |
+          When `v-model.numner` is used, Vue parses the value into a JavaScript
+          `Number` type.
+      correct: b
+
+extend_v_model_custom_component:
+  question: |
+    We in the process of building a nice checkout form for our pizza customers.
+    It looks something like this:
+
+    ```vue
+    <template>
+        <form>
+            <p>Please, enter your info here:</p>
+
+            <fancy-input
+                id="name"
+                v-model="client.name"
+                label="Name:"
+            />
+        </form>
+    </template>
+    ```
+
+    We have just written the first input for the customer's name and we want it
+    to be a custom component but we've forgotten how to implement `v-model` in
+    a custom component in Vue 2!
+
+    Which of these templates would do the trick, assuming our component has a prop
+    `value` of type `String`?
+  explanation: |
+    In vue 2, `v-model` translates to `:value` and an `$emit` with the event
+    `input` and the text to be changed! Just remember that in Vue 3, the
+    default prop to be used in `v-model` is `modelValue`!
+
+    > Alternatively, you can also pass `v-model:value` in Vue 3. The `:value` part
+    > acts as a parameter of the `v-model` *"function"* and tells it that
+    > the prop we want to use is called `value`. Just like in Vue 2!
+  features:
+    multiple_choice:
+      choices:
+        a: |
+          ```vue
+          <input
+              :id="id"
+              :name="id"
```

Comment:
We should take off the `id` and `name` - it's unnecessary. But we can keep the `type` to make it more realistic.